server/src/main/java/io/crate/protocols/postgres/PgClient.java

```diff
@@ -0,0 +1,655 @@
+            // We set that property earlier in `connect()` to the pg-client channel.
+            // This lets the tcp-transport communicate via the channel established here in the PgClient.
+
+            var handler = new Netty4MessageChannelHandler(pageCacheRecycler, transport);
+            channel.pipeline().addLast("dispatcher", handler);
+            Netty4TcpChannel tcpChannel = channel.attr(Netty4Transport.CHANNEL_KEY).get();
+
+            ActionListener<Version> onHandshakeResponse = ActionListener.wrap(
+                version -> {
+                    var connection = new TunneledConnection(
+                        transport.outboundHandler(),
+                        node,
+                        tcpChannel,
+                        profile,
+                        version
+                    );
+                    long relativeMillisTime = this.transport.getThreadPool().relativeTimeInMillis();
+                    tcpChannel.getChannelStats().markAccessed(relativeMillisTime);
+                    tcpChannel.addCloseListener(ActionListener.wrap(connection::close));
+                    result.complete(connection);
+                },
+                e -> {
+                    var error = e instanceof ConnectTransportException ? e : new ConnectTransportException(node, "general node connection failure", e);
+                    try {
+                        CloseableChannel.closeChannels(List.of(tcpChannel), false);
+                    } catch (Exception ex) {
+                        error.addSuppressed(ex);
+                    } finally {
+                        result.completeExceptionally(error);
+                    }
+                }
+            );
+            transport.executeHandshake(node, tcpChannel, profile, onHandshakeResponse);
+        }
+
+        private void handleParameterStatus(ByteBuf msg) {
+            PostgresWireProtocol.readCString(msg); // consume name
+            PostgresWireProtocol.readCString(msg); // consume value
+        }
+
+        private void handleAuth(Channel channel, ByteBuf msg) {
+            AuthType authType = AuthType.of(msg.readInt());
+            if (LOGGER.isTraceEnabled()) {
+                LOGGER.trace("Server sent authentication request type={}", authType);
+            }
+            switch (authType) {
+                case OK:
+                    break;
+
+                case CLEARTEXT_PASSWORD:
+                    ByteBuf buffer = channel.alloc().buffer();
+                    ClientMessages.writePasswordMessage(buffer, connectionInfo.password());
+                    channel.writeAndFlush(buffer);
+                    break;
+
+                default:
+                    throw new UnsupportedOperationException(authType + " is not supported");
+            }
+        }
+    }
+
+    enum AuthType {
+        OK,
+        KERBEROS,
```

Comment:
Is this used somewhere? (same for MD5_Password) exists to fulfill the protocol spec?